index.d.ts

```diff
@@ -0,0 +1,138 @@
+     */
+    ignoreCaret?: boolean;
+    /**
+     * Some images has difference while comparing because of antialiasing.
+     * These diffs will be ignored by default. You can use ignoreAntialiasing option with false value to disable ignoring such diffs.
+     * In that way antialiased pixels will be marked as diffs.
+     */
+    ignoreAntialiasing?: boolean;
+    /**
+     * Sometimes the antialiasing algorithm can work incorrectly due to some features of the browser rendering engine.
+     * Use the option antialiasingTolerance to make the algorithm less strict.
+     * With this option you can specify the minimum difference in brightness (zero by default)
+     * between the darkest/lightest pixel (which is adjacent to the antialiasing pixel) and theirs adjacent pixels.
+     *
+     * We recommend that you don't increase this value above 10. If you need to increase more than 10 then this is definitely not antialiasing.
+     */
+    antialiasingTolerance?: number;
+}
+
+/**
+ * The options passed to looksSame.createDiff function without diff
+ */
+interface CreateDiffAsBufferOptions {
+    /**
+     * The baseline image path
+     */
+    reference: string;
+    /**
+     * The current image path
+     */
+    current: string;
+    /**
+     * Color to highlight the differences
+     * e.g. '#ff00ff'
+     */
+    highlightColor: string;
+    /**
+     * strict comparsion
+     */
+    strict: boolean;
+    /**
+     * ΔE value that will be treated as error in non-strict mode
+     */
+    tolerance: number;
+}
+
+/**
+ * The options passed to looksSame.createDiff function
+ */
+interface CreateDiffOptions extends CreateDiffAsBufferOptions {
+    /**
+     * The diff image path to store
+     */
+    diff: string;
+}
+
+/**
+ * Pass to looksSame.colors function
+ */
+interface LooksSameColor {
+    /**
+     * Red
+     */
+    R: number;
+    /**
+     * Green
+     */
+    G: number;
+    /**
+     * Blue
+     */
+    B: number;
+}
+
+/**
+ * Compare two images with options
+ * @param image1 The first image path
+ * @param image2 The second image path
+ * @param options The options passed to looksSame function
+ * @param callback Call when finish compare
+ */
+declare function looksSame(image1: string, image2: string, options: LooksSameOptions, callback: LooksSameCallback): void;
+/**
+ * Compare two images
+ * @param image1 The first image path
+ * @param image2 The second image path
+ * @param callback Call when finish compare
+ */
+declare function looksSame(image1: string, image2: string, callback: LooksSameCallback): void;
+
+// https://stackoverflow.com/questions/44058101/typescript-declare-third-party-modules
+declare namespace looksSame {
+    function createDiff(options: CreateDiffOptions, callback: (error: Error | null) => any): void;
+    function createDiff(options: CreateDiffAsBufferOptions, callback: (error: Error | null, buffer: Buffer) => any): void;

```

Comment:
To my mind we don't need this overload. Do you agree?

Reply:
With this overload, typescript can provide stronger type infer like this:

![image](https://user-images.githubusercontent.com/5453359/48424899-a8582c80-e79e-11e8-829f-464b7d2d1a46.png)

![image](https://user-images.githubusercontent.com/5453359/48424926-b8700c00-e79e-11e8-8b1f-287290789d22.png)